### PR TITLE
fix(files): add visible label to webdav URL in settings

### LINKED ISSUE #41835: [BITV] 9.1.3.1h/3.1 - Modal dialog "File settings" - The input field for the WebDAV URL does not have a label associated with it programmatically.  (2)
It is recommended to implement a visual label (e.g., "WebDAV Url") using the "label" element, which would be linked to the field using the "for" and "id" attributes.

https://report.bitvtest.de/default-en/cbedb5ef-4815-4cba-a83d-995599c178c3.html#checkpoint-fa5c8d46f9-v3-n1

9.1.3.1h/11.1 - https://report.bitvtest.de/default-en/cbedb5ef-4815-4cba-a83d-995599c178c3.html#checkpoint-fa5c8d46f9-v11-n1

### PR DESCRIPTION
**Pull request by ShGKme**
* Resolves: #41835

Before | After
---|---
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by susnux — APPROVED** [review]

File: apps/files/src/views/Settings.vue [hunk 1]
```diff
@@ -56,6 +56,7 @@
 		<!-- Webdav URL-->
 		<NcAppSettingsSection id="webdav" :name="t('files', 'WebDAV')">
 			<NcInputField id="webdav-url-input"
```
Comment (susnux) [inline apps/files/src/views/Settings.vue:58]: Should it not be NcTextField?

**Review by Pytal — APPROVED** [review]
Other than the NcTextField suggestion

File: apps/files/src/views/Settings.vue (hunk 1 above)
Comment (ShGKme) [reply apps/files/src/views/Settings.vue:58]: Then we need to allow passing any content to the trailing button slot.

https://github.com/nextcloud-libraries/nextcloud-vue/blob/79e68856175859975dc6b7233eb8d503d37f5a6a/src/components/NcTextField/NcTextField.vue#L141-L145

**Comment by ShGKme** [comment]
Rebased onto master to resolve conflict

**Review by marcoambrosini — APPROVED** [review]

File: apps/files/src/views/Settings.vue (hunk 1 above)
Comment (susnux) [reply apps/files/src/views/Settings.vue:58]: Yes I also misuse NcInputField for some projects because of that limitation.

**Pull request merged**